packages/aws-cdk/test/api/deploy-stack.test.ts

```diff
@@ -1127,30 +1127,37 @@ describe('disable rollback', () => {
 });
 
 test.each([
-  [StackStatus.UPDATE_FAILED, 'failpaused-need-rollback-first'],
-  [StackStatus.CREATE_COMPLETE, 'replacement-requires-norollback'],
-])('no-rollback and replacement is disadvised: %p -> %p', async (stackStatus, expectedType) => {
+  [StackStatus.UPDATE_FAILED, 'no-rollback', 'no-replacement', 'did-deploy-stack'],
+  [StackStatus.UPDATE_FAILED, 'no-rollback', 'replacement', 'failpaused-need-rollback-first'],
+  [StackStatus.UPDATE_FAILED, 'rollback', 'replacement', 'failpaused-need-rollback-first'],
+  [StackStatus.UPDATE_COMPLETE, 'no-rollback', 'replacement', 'replacement-requires-rollback'],
```

Comment:
I haven't traced through the code fully, but this seems to read that `replacement-requires-rollback`, will successfully complete when a replacement is needed but we passed in `no-rollback`.

Reply:
When the stack is in a stable state (`UPDATE_COMPLETE`), the user passes `--no-rollback`, and there is a `replacement` to do, the `deployStack` function will say: `replacement-requires-rollback`.

That is to say, the combination of `replacement` and `no-rollback` is not possible. This is a CloudFormation limitation.

What will happen in practice is that our CLI will catch this return code and ask "Do you want to perform a regular (--rollback) deployment (y/n)?"